src/main/java/ui/GUIController.java

```diff
@@ -102,13 +108,38 @@ public void panelFilterExpressionChanged(FilterPanel changedPanel) {
                 .collect(Collectors.toList());
     }
 
+    private void updateRateLimitsEvent(UpdateRateLimitsEvent e) {
+
+        updateAPIBox(e);
+
+        if (e.allowUpdateOfRefreshRate) {
+            updateSyncRefreshRate(e);
+        }
+
+    }
+
+    private void updateSyncRefreshRate(UpdateRateLimitsEvent e) {
```

Comment:
This method seems to be doing two things: calculate refresh rate, update it on UI